### ISSUE TITLE
Get rid of ``asyncCheck``

### ISSUE THREAD

**Issue by dom96** [nim-lang/Nim#4077]
An idea popped into my head just now. We might be able to get rid of the need for ``asyncCheck``, by
simply defaulting every Future's callback to the one set by ``asyncCheck``. Although I'm not sure what consequences that may have.

**Comment by zielmicha** [comment]
This won't work - consider the following example:
```
let a = thing1()
await thing2()
# here `a` completes, calling callback from asyncCheck
await a # but souldn't
```

**Comment by dom96** [comment]
Indeed. Maybe we could make use of destructors? When `a` goes out of scope without being awaited it gets checked for errors.

**Comment by Yardanico** [comment]
This can be revisited again since @Araq makes new destructors
